rotary_utils/rotate.py

```diff
@@ -0,0 +1,286 @@
+    """
+
+    sequence_length = len(sequence_record.seq)
+    if rotate_position > sequence_length:
+        logger.error(f'Desired rotate position ({rotate_position}) is larger than the sequence length '
+                     f'({sequence_length} bp).')
+        raise RuntimeError
+
+    elif rotate_position == sequence_length:
+        logger.warning(f'Desired rotate position ({rotate_position}) is equal to the sequence length '
+                       f'({sequence_length} bp), so the contig will effectively not be rotated.')
+
+    elif rotate_position < 0:
+        logger.debug(f'Desired rotate position ({rotate_position}) is less than 0.')
+        raise RuntimeError
+
+    logger.debug(f'Rotating sequence to start at position: {rotate_position}')
+    contig_sequence_rotated_front = sequence_record.seq[rotate_position:sequence_length]
+    contig_sequence_rotated_back = sequence_record.seq[0:rotate_position]
+    contig_sequence_rotated = contig_sequence_rotated_front + contig_sequence_rotated_back
+
+    # Update SeqRecord
+    sequence_record.seq = contig_sequence_rotated
+
+    if strip_description:
+        sequence_record.description = sequence_record.name
+
+    return sequence_record
+
+
+def rotate_sequence_to_fraction(sequence_record: SeqIO.SeqRecord, rotate_fraction: float,
+                                strip_description: bool = True):
+    """
+    Rotates an input (circular) sequence to a specified fraction of the total length
+    :param sequence_record: BioPython SeqRecord object containing the sequence to rotate
+    :param rotate_fraction: fraction of the contig length to rotate the contig to (rounded down to nearest bp)
+    :param strip_description: boolean of whether to trim off the read description in the output sequences
+    :return: BioPython SeqRecord object of the rotated FastA file
+    """
+
+    sequence_length = len(sequence_record.seq)
+
+    if rotate_fraction < 0:
+        logger.error(f'Requested rotation fraction ({rotate_fraction}) is less than 0.')
+        raise RuntimeError
+
+    else:
+        rotate_position = math.floor(sequence_length * rotate_fraction)
+
+    sequence_record = rotate_sequence(sequence_record, rotate_position=rotate_position,
+                                      strip_description=strip_description)
+
+    return sequence_record
+
+
+def rotate_sequence_to_midpoint(sequence_record: SeqIO.SeqRecord, strip_description: bool = True):
+    """
+    Rotates an input (circular) sequence to the approximate midpoint
+    :param sequence_record: BioPython SeqRecord object containing the sequence to rotate
+    :param strip_description: boolean of whether to trim off the read description in the output sequences
+    :return: BioPython SeqRecord object of the rotated FastA file
+    """
+
+    sequence_record = rotate_sequence_to_fraction(sequence_record, rotate_fraction=0.5,
+                                                  strip_description=strip_description)
+
```

Comment:
@jmtsuji One could probably remove the blank line here and not lose any readability.

Reply:
Done